core/src/main/scala/com/databricks/labs/remorph/parsers/IncompleteParser.scala

```diff
@@ -13,4 +15,40 @@ trait IncompleteParser[T] extends ParseTreeVisitor[T] {
       case x => x
     }
   }
+  // TODO: Merge formatContext with visitChildren to consistent formatting.
+  /**
+   * Formats the text of a given `RuleContext` by concatenating the text of all its children nodes,
+   * separated by spaces.
+   *
+   * @param ctx the `RuleContext` to format.
+   * @return a `String` representing the concatenated text of all children nodes, separated by spaces.
+   *         If the context has no children, returns an empty string.
+   */
+  def formatContext(contexts: RuleContext): ir.UnresolvedCommand = {
+    def formatContextRecursive(ctx: RuleContext): String = {
+      if (ctx.getChildCount == 0) {
+        ctx.getText
+      } else {
+        val builder = new StringBuilder
+        for (i <- 0 until ctx.getChildCount) {
+          val child = ctx.getChild(i)
+          if (child.isInstanceOf[RuleContext]) {
+            builder.append(formatContextRecursive(child.asInstanceOf[RuleContext]))
+          } else {
+            builder.append(child.getText)
+          }
+          builder.append(" ")
+        }
+        // remove any trailing spaces and some regular expression
+        // TODO Fix this to not include regex rather check based on Generic String Context Expression.
+        builder.toString
```

Comment:
I have already implemented retrieval of the original input text given a context - however it is not checked in yet:

```scala
package com.databricks.labs.remorph.utils

import org.antlr.v4.runtime.misc.Interval

object ParsingUtils {

  /**
   * Creates a string representation of the text represented by the given ANTLR ParserRuleContext.
   *
   * Note that this should exactly reflect the original input text as bounded by the source interval
   * recorded by the pareser.
   */
  def getTextFromParserRuleContext(ctx: org.antlr.v4.runtime.ParserRuleContext): String =
    ctx.start.getInputStream.getText(new Interval(ctx.start.getStartIndex, ctx.stop.getStopIndex))
}
```

Implementing the same for say two tokens is also trivial. This way you get back the exact input text. There are more nuances to this such as child terminal nodes and error nodes. It is best to retrieve from the input.

I will split that utility into its own PR and submit it.